llpc/translator/lib/SPIRV/libSPIRV/SPIRVEntry.cpp

```diff
@@ -275,6 +275,14 @@ bool SPIRVEntry::hasDecorate(Decoration Kind, size_t Index,
   return true;
 }
 
+const char *SPIRVEntry::getDecorateString(Decoration kind) const {
+  DecorateMapType::const_iterator loc = Decorates.find(kind);
+  if (loc == Decorates.end())
+    return "";
```

Comment:
Is this correct? I don't know much about `OpDecorateString`, but shouldn't you be able to differentiate between an empty-string decoration and not having a decoration at all?